htsinfer/infer_adapter.py

```diff
@@ -0,0 +1,203 @@
+        factor (float) : factor by which first adapter is greater than the
+        second adapter.
+
+    Returns:
+        Type of Adapter that is present in library.
+    """
+    # Process file 1
+    LOGGER.debug(f"Processing file 1: {file_1}")
+    result_1 = process_fastq_file(file_1, max_records, min_match, factor)
+
+    # Process file 2
+    result_2 = "not_available"
+    if file_2:
+        LOGGER.debug(f"Processing file 2: {file_2}")
+        result_2 = process_fastq_file(file_2, max_records, min_match, factor)
+
+    LOGGER.debug("Returning results...")
+    return (result_1, result_2)
+
+
+def process_fastq_file(
+    file: str,
+    max_records: int = 10000,
+    min_match: float = 10,
+    factor: float = 2
+) -> str:
+    """Process adapters count info.
+
+    Args:
+        file (str) : File path to read/first mate library.
+        max_records: Limit processing to the indicated number of records,
+            starting from the first record.
+        min_match (float) : minimum match percentage that first adapter needs
+        to have.
+        factor (float) : factor by which first adapter is greater than the
+        second adapter.
+
+    Returns:
+        Adapter information
+    """
+    _open = partial(
+        gzip.open, mode='rt'
+    ) if file.endswith(".gz") else open
+
+    try:
+        LOGGER.debug("Opening file...")
+        with _open(file) as _file:  # type: ignore
+
+            adapter_counts: Dict[str, float] = {}
+            records: int = 0
+            total_count: int = 0
+            LOGGER.debug("Processing reads")
+            for record in FastqGeneralIterator(source=_file):
+                # Get next read
+                read = record[1]
+
+                # Searching for adapters in read
+                for _, (_, key) in TRIE.iter(read):
+                    if key in adapter_counts:
+                        adapter_counts[key] += 1
+                    else:
+                        adapter_counts[key] = 1
+                    total_count += 1
+
+                # Update records
+                records += 1
+
+                # End if requested records processed
+                if max_records and records >= max_records:
+                    break
+
+            if total_count == 0:
+                return "NA"
+
+            # Calculating Percentage
+            for i in adapter_counts:
+                adapter_counts[i] = round(
+                    (adapter_counts[i]/total_count)*100, 2
+                    )
+
+            # Converting dictionary into dataframe
+            adapters_df = pd.DataFrame(adapter_counts.items())
+            adapters_df.columns = ['Adapter', 'Count %']
+            adapters_df = adapters_df.sort_values(
+                by='Count %', ascending=False
+                ).reset_index(drop=True)
+            LOGGER.debug(f"Creating {file}_adapters_count.csv")
+            adapters_df.to_csv(f"{file}_adapters_count.csv")
+
+            # Checking confidence score
```

Comment:
I think confidence is not a good term to use in the actual code as it kinda implies its use in math, and we are certainly not calculating any confidence. In fact, the same is true for inference, we are not inferring anything, we are just counting :)